# 2. Chuẩn bị vấn đề

## 2.1 Khai báo thư viện

In [14]:
import pandas as pd
from IPython import display

## 2.2 Nạp dữ liệu

In [15]:
colunms_name = ["sepal_length", "sepal_width", "petal_length", "petal_width", "class"]
data = pd.read_csv("iris/iris.csv", names = colunms_name)

data.head()

sepal_length  sepal_width  petal_length  petal_width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

# 3. Phân tích dữ liệu

## 3.1 Thống kê mô tả

### (1) Hiển thị một số thông tin về dữ liệu
+ Số dòng, số cột của dữ liệu
+ Kiểu dữ liệu của các cột
+ 5 dòng đầu và 5 dòng cuối của dữ liệu
+ Thông tin về dữ liệu

In [17]:
print("Shape: ", data.shape)
print("Data type: ", data.dtypes)
print("Contents: ")
display.display(data.head())
display.display(data.tail())
print("Info: ")
data.info()

Shape:  (150, 5)
Data type:  sepal_length    float64
sepal_width     float64
petal_length    float64
petal_width     float64
class            object
dtype: object
Contents: 


sepal_length  sepal_width  petal_length  petal_width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa

sepal_length  sepal_width  petal_length  petal_width           class
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

Info: 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   class         150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


#### Nhận xét
+ Dữ liệu có 4 tính chất để phân lớp: sepal-width, sepal-length, petal-width và petal-width.
+ Dữ liệu của 4 tính chất được tính cùng đơn vị đo.
+ Tổng số dữ liệu là 150 dòng.
+ Dữ liệu phân lớp ở cột class.

### (2) Kiểm tra tính toàn vẹn của dữ liệu.
+ 